crates/services/p2p/src/peer_manager.rs

```diff
@@ -160,6 +186,41 @@ impl PeerManagerBehaviour {
     pub fn get_peer_id_with_height(&self, height: &BlockHeight) -> Option<PeerId> {
         self.peer_manager.get_peer_id_with_height(height)
     }
+
+    pub fn report_peer(
+        &mut self,
+        peer_id: PeerId,
+        peer_score: PeerScore,
+        reporting_service: &str,
+    ) -> Option<PeerScore> {
+        if let Some(latest_peer_score) = self
+            .peer_manager
+            .update_peer_score_with(peer_id, peer_score)
+        {
+            info!(target: "fuel-p2p", "{reporting_service} updated {peer_id} with new score {latest_peer_score}");
+
+            Some(latest_peer_score)
+        } else {
+            None
+        }
+    }
+
+    pub fn report_gossip_score(&mut self, peer_id: &PeerId, gossip_score: f64) {
+        if self
+            .peer_manager
+            .non_reserved_connected_peers
+            .contains_key(peer_id)
+            && gossip_score
+                < self
+                    .peer_manager
+                    .peer_score_config
+                    .get_min_gossipsub_score()
+        {
+            self.peer_manager
```

Comment:
After today's discussion, we agreed that it is worth considering moving the `PeerManager` logic above to the `P2PService` level or upper to disconnect immediately. Also, it will reduce the exact method duplication on each level(like triple `report_peer` and so on).